File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03829/p03829_num2_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, A, B;
    cin >> N >> A >> B;

    vector<int> X(N);
    for (int i = 0; i < N; i++) {
        cin >> X[i];
    }

    int total_fatigue = 0;
    for (int i = 0; i < N - 1; i++) {
        int distance = X[i + 1] - X[i];
        int teleport_fatigue = distance * B;
        int walk_fatigue = distance * A;
        total_fatigue += min(teleport_fatigue, walk_fatigue);
    }

    cout << total_fatigue << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03829/p03829_num2_parsed.py
import sys

def calculate_min_fatigue(N, A, B, X):
    # calculate minimum possible total increase of fatigue level
    min_fatigue = 0
    
    for i in range(N-1):
        distance = X[i+1] - X[i]
        
        # calculate fatigue level for walking
        walking_fatigue = A * distance
        
        # calculate fatigue level for teleporting
        teleport_fatigue = B
        
        # choose minimum fatigue level between walking and teleporting
        min_fatigue += min(walking_fatigue, teleport_fatigue)
    
    return min_fatigue

# read input from STDIN
input_line = sys.stdin.readline().strip()
N, A, B = map(int, input_line.split())

input_line = sys.stdin.readline().strip()
X = list(map(int, input_line.split()))

# calculate and print the minimum possible total increase of fatigue level
print(calculate_min_fatigue(N, A, B, X))